### DISCUSSION TITLE
issues with png export

### DISCUSSION THREAD

**Discussion by dark-m0de** [rlespinasse/docker-drawio-desktop-headless#91] [Q&A]
Hi @rlespinasse,

thank you for providing this project. It was very helpful for me and inspired me to create my own version (to be used as a runner image in GitLab).  Basically, I use the same commands in the dockerfile, just without entrypoint / cmd in the end.

Instead I have a `drawio` file with the following content that I put into `/usr/local/bin`
```sh
#!/usr/bin/env bash
set -euo pipefail

if [[ "${SUPPRESS_UNWANTED_OUTPUT:?}" == "true" ]]; then
  xvfb-run -a /opt/drawio/drawio "$@" --no-sandbox --disable-gpu 2>&1 | grep -Fvf "/opt/drawio-desktop/unwanted-output-lines.txt"
else
  xvfb-run -a /opt/drawio/drawio "$@" --no-sandbox --disable-gpu 2>&1
fi
```

While I can use this image inside my CI pipeline to call commands like `drawio --help` or to export PDF files, the png export seems to have problems.
The command `drawio -x -o flow.png flow.drawio` fails with the following output:

```
[212:0522/055006.581379:ERROR:validation_errors.cc(117)] Invalid message: VALIDATION_ERROR_DESERIALIZATION_FAILED
[212:0522/055006.581435:ERROR:interface_endpoint_client.cc(702)] Message 2131977360 rejected by interface viz.mojom.CopyOutputResultSender
[212:0522/055006.581524:ERROR:browser_child_process_host_impl.cc(754)] Terminating child process for bad message: Received bad user message: Validation failed for viz.mojom.CopyOutputResultSender.0  [VALIDATION_ERROR_DESERIALIZATION_FAILED]
Error: Export failed: flow.drawio
```

Did you, by any chance, once have a similar error and know how to fix this?

**Comment by dark-m0de** [comment]
P.S.: I also tried this, but had the same result

```sh
export DISPLAY=":42"
Xvfb ":42" -nolisten unix & /opt/drawio/drawio -x -o /opt/drawio-desktop/flow.png /opt/drawio-desktop/flow.drawio --no-sandbox --disable-gpu
```

**Comment by rlespinasse** [comment]
Hello,

Can you produce a test case GitLab repository snippet to help reproduce your issue?

On the drawio-export docker image (built on top of this docker image), someone uses it in GitLab.
Maybe the related issue snippets can help you: https://github.com/rlespinasse/drawio-export/issues/82
